### PR TITLE
Hostdb Improvements

### PR DESCRIPTION
**Pull request by DavidVorick**
A logger was added, so far just a single logging statment.

A severe bug in the hostdb construction was fixed involving pointers.

### CODE REVIEW

File: modules/hostdb/hostdb.go [hunk 1]
```diff
@@ -26,16 +23,17 @@ const (
...
+	errNilConsensusSet = errors.New("consensus set cannot be nil")
 )
 
 // The HostDB is a database of potential hosts. It assigns a weight to each
 // host based on their hosting parameters, and then can select hosts at random
 // for uploading files.
 type HostDB struct {
-	consensusSet *consensus.ConsensusSet
-	gateway      modules.Gateway
+	consensusSet    *consensus.ConsensusSet
```
Comment (lukechampine) [inline modules/hostdb/hostdb.go]: `modules.ConsensusSet`?

**Comment by lukechampine** [comment]
looks good

File: modules/hostdb/hostdb.go (hunk 1 above)
Comment (DavidVorick) [inline modules/hostdb/hostdb.go]: oh yeah might as well fix that

**Comment by DavidVorick** [comment]
ok, should be all better

**Pull request merged**